### PR TITLE
Switch to karma for test running

### PR DESCRIPTION
**Pull request by devpow112**
This fixes issues with test running. This also removes the CI build as it's not needed and fixes some eslint issues. Working towards #106.

Recommend viewing with white-space off
https://github.com/Brightspace/awards-leaderboard-ui/pull/127/files?diff=split&w=1

### CODE REVIEW

**Review by cbelan-d2l — COMMENTED** [review]
Thank you! Thank you !!! 
This is amazing 👍 🥇

File: README.md [hunk 1]
```diff
@@ -45,7 +45,7 @@ There are tests with this repo that run within CI. To test them locally within a
 browser, run
 
 ```
-polymer test --skip-plugin sauce -p
+npm test
```
Comment (cbelan-d2l) [inline README.md:48]: Hooray

File: ci-script.sh [hunk 2]
```diff
@@ -15,8 +15,5 @@ if [ $TRAVIS_PULL_REQUEST != false ] && [ $TRAVIS_SECURE_ENV_VARS == true ]; the
     npm run test:sauce || travis_terminate 1;
 else
     echo "Not a pull request and/or no secure environment variables, running headless tests...";
-    npm run test:travis || travis_terminate 1;
+    npm run test:local || travis_terminate 1;
 fi
-
-echo 'Build'
-npm run build
```
Comment (cbelan-d2l) [inline ci-script.sh:22]: Don't need to run the build anymore?

Comment (devpow112) [reply ci-script.sh:22]: The build isn't used so it's not needed. This component is included in BSI which itself builds everything. Eventually we can switch to `es-dev-server` which will eliminate the need for polymer and the build fully.

**Review by cbelan-d2l — APPROVED** [review]

**Pull request merged**